code/modules/surgery/organs/organ_internal.dm

```diff
@@ -0,0 +1,314 @@
+#define PROCESS_ACCURACY 10
+
+/obj/item/organ/internal
+	origin_tech = "biotech=2"
+	force = 1
+	w_class = 2
+	throwforce = 0
+	var/zone = "chest"
+	var/slot
+	vital = 0
+	var/organ_action_name = null
+
+/obj/item/organ/internal/proc/insert(mob/living/carbon/M, special = 0)
+	if(!iscarbon(M) || owner == M)
+		return
+
+	var/obj/item/organ/internal/replaced = M.get_organ_slot(slot)
+	if(replaced)
+		replaced.remove(M, special = 1)
+
+	owner = M
+	M.internal_organs |= src
+	loc = null
```

Comment:
Ah, I think I see code like that in shadowling code, where they drop all their equipment upon transforming.

Couldn't you define a proc for carbons to drop all possessions, and have the "snowflake" be in there, so that you don't have organ code messing with shadowling code, and whatever else needs it?